compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -761,15 +778,15 @@ fn print_crate_info(
                 };
                 let t_outputs = rustc_interface::util::build_output_filenames(attrs, sess);
                 let id = rustc_session::output::find_crate_name(sess, attrs);
-                if *req == PrintRequest::CrateName {
-                    safe_println!("{id}");
-                    continue;
-                }
-                let crate_types = collect_crate_types(sess, attrs);
-                for &style in &crate_types {
-                    let fname =
-                        rustc_session::output::filename_for_input(sess, style, id, &t_outputs);
-                    safe_println!("{}", fname.as_path().file_name().unwrap().to_string_lossy());
+                if req.kind == CrateName {
```

Comment:
Even though this PR didn't introduce this, can you maybe create two separate match arms for `FileNames` and `CrateName`? This does introduce a bit of code duplication, but we wouldn't have the `build_output_filenames` in the `CrateName` arm and this seems a little cleaner overall imo.